src/typing/flow_js.ml

```diff
@@ -1481,13 +1481,15 @@ let rec __flow cx (l, u) trace =
     (* The sink component of an annotation constrains values flowing
        into the annotated site. *)
 
-    | _, AnnotT (sink_t, _) ->
+    | l, AnnotT (sink_t, _)
+    | ShiftT(l), ShiftT(AnnotT (sink_t, _)) ->
```

Comment:
This and a few other sites have been defering to the generic `ShiftT~>ShiftT => unify` case.  My operating assumption is that Flow has been using the `Node.utils` model of inheritence (no prototype chaining on the classes themselves, just the instances), so everything tends to work (as long as you don't try using `Function` methods of a class inside the body of a function, for instance).  If I'm trampling on some important invariants here, I'd appreciate some feedback.